templates/commands/goldentest/verify.go

```diff
@@ -60,9 +68,127 @@ func (c *VerifyCommand) Run(ctx context.Context, args []string) error {
 		return fmt.Errorf("failed to parse flags: %w", err)
 	}
 
-	if _, err := parseTestCases(ctx, c.flags.Location, c.flags.TestName); err != nil {
-		return fmt.Errorf("failed to parse golden test: %w", err)
+	testCases, err := parseTestCases(ctx, c.flags.Location, c.flags.TestName)
+	if err != nil {
+		return fmt.Errorf("failed to parse golden tests: %w", err)
+	}
+
+	// Create a temporary directory to render golden tests
+	tempDir, err := renderTestCases(testCases, c.flags.Location)
+	defer os.RemoveAll(tempDir)
+	if err != nil {
+		return fmt.Errorf("failed to render test cases: %w", err)
 	}
 
-	return fmt.Errorf("unimplemented")
+	var merr error
+	// Highlight error message color, given diff text might be hundreds lines long.
+	red := color.New(color.FgRed).SprintFunc()
+	green := color.New(color.FgGreen).SprintFunc()
+	resultReport := "\nTest Report:\n"
+
+	for _, tc := range testCases {
+		goldenDataDir := filepath.Join(c.flags.Location, goldenTestDir, tc.TestName, testDataDir)
+		tempDataDir := filepath.Join(tempDir, goldenTestDir, tc.TestName, testDataDir)
+
+		fileSet := make(map[string]struct{})
+		if err := addTestFile(&fileSet, goldenDataDir); err != nil {
+			return err
+		}
+		if err := addTestFile(&fileSet, tempDataDir); err != nil {
+			return err
+		}
+
+		dmp := diffmatchpatch.New()
+
+		var tcErr error
+		for relPath := range fileSet {
+			goldenFile := filepath.Join(goldenDataDir, relPath)
+			tempFile := filepath.Join(tempDataDir, relPath)
+
+			goldenContent, err := os.ReadFile(goldenFile)
+			if err != nil {
+				if errors.Is(err, os.ErrNotExist) {
+					failureText := red(fmt.Sprintf("-- [%s] generated, however not recorded in test data", goldenFile))
+					err := fmt.Errorf(failureText)
+					tcErr = errors.Join(tcErr, err)
+					continue
+				} else {
+					return fmt.Errorf("failed to read (%s): %w", goldenFile, err)
+				}
+			}
+
+			tempContent, err := os.ReadFile(tempFile)
+			if err != nil {
+				if errors.Is(err, os.ErrNotExist) {
+					failureText := red(fmt.Sprintf("-- [%s] expected, however missing", goldenFile))
+					err := fmt.Errorf(failureText)
+					tcErr = errors.Join(tcErr, err)
+					continue
+				} else {
```

Comment:
Since the previous `if` block has a continue, the convention is to avoid the unnecessary "else" here. So instead of `else {foo}` it can just be `foo` . Same at line 116.

Reply:
Done.